src/main/java/org/testng/internal/BaseTestMethod.java

```diff
@@ -133,10 +115,8 @@ public ITestClass getTestClass() {
    */
   @Override
   public void setTestClass(ITestClass tc) {
-    assert null != tc;
-    if (! tc.getRealClass().equals(m_method.getDeclaringClass())) {
-      assert m_method.getDeclaringClass().isAssignableFrom(tc.getRealClass()) :
-        "\nMISMATCH : " + tc.getRealClass() + " " + m_method.getDeclaringClass();
+    if (tc == null) {
+      throw new IllegalArgumentException("test class cannot be null");

```

Comment:
You missed a check on the type of classes

Reply:
@juherr - I dont think that check was working. Moreover its an `assert` which by default is not enabled in most of the execution. That was why I got rid of it.

Reply:
Ok, why not if it doesn't break tests 😉

Reply:
@juherr - It was breaking tests 😁 That was why I chose to get rid of it.

Reply:
Ok, so I hope the idea behind the assertion is deprecated now and we are not hiding a bug.

Reply:
@juherr - On second thoughts, I went back and fixed this as well to add edit checks which were supposed to have been covered by the `assert` statements (Which will only get enabled when user adds the `-ea` flag to the JVM. Not sure how may test executions have even been doing that till now). I have amended the commit and pushed in the changes.